### ISSUE TITLE
[BUG][Fuzzer][wrong-code] reduce_max/reduce_min with clear=False silently ignores nan_propagate=True (NaN dropped in write-back merge)

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2697]
## Describe the bug

`T.reduce_max`/`T.reduce_min` with `clear=False` and `nan_propagate=True` silently drops NaN: the reduction returns the pre-existing accumulator value instead of the documented NaN, because the `clear=False` write-back merge (`MakeUpdate`) emits a plain `__hmax`/`__hmin` and never consults `op.nan_propagate`.

## Expected behavior

`nan_propagate=True` should propagate NaN through the reduction regardless of `clear`. The docstring guarantees this unconditionally ([reduce_op.py L156-159](https://github.com/tile-ai/tilelang/blob/cd37ed5f/tilelang/language/reduce_op.py#L156-L159): *"When True, NaN inputs propagate through the reduction (CUDA `__hmax_nan`)"*; [L180-182](https://github.com/tile-ai/tilelang/blob/cd37ed5f/tilelang/language/reduce_op.py#L180-L182) for `reduce_min`) — neither restricts the guarantee to `clear=True`. The project's own runtime test asserts `nan_propagate=True` yields NaN ([test_tilelang_language_reduce_maxmin_nan.py L89, L103](https://github.com/tile-ai/tilelang/blob/cd37ed5f/testing/python/language/test_tilelang_language_reduce_maxmin_nan.py#L89-L103)), so `out` should be NaN on the `clear=False` path too.

## Reproducible example

Seed `out_frag[0] = -1.0`, put a NaN in the reduce dim, reduce with `clear=False, nan_propagate=True`. Expected NaN; actual `-1.0`. The `clear=True` positive control (same script) confirms the flag works on the `MakeReduce` path.

```python
import torch, tilelang, tilelang.language as T

def make(length, dtype, clear, nanp, fn):
    @T.prim_func
    def kernel(a: T.Tensor((length,), dtype), out: T.Tensor((1,), dtype)):
        with T.Kernel(1, threads=32):
            frag = T.alloc_fragment((length,), dtype)
            of = T.alloc_fragment((1,), dtype)
            T.copy(a, frag)
            of[0] = -1.0                      # seed pre-existing accumulator
            fn(frag, of, clear=clear, nan_propagate=nanp)
            T.copy(of, out)
    return kernel

length = 64
for dtype, tdt in [("float16", torch.float16), ("bfloat16", torch.bfloat16)]:
    a = torch.arange(length, dtype=torch.float32).to(tdt).cuda()
    a[7] = float("nan")
    for clear in (True, False):
        k = tilelang.compile(make(length, dtype, clear, True, T.reduce_max))
        out = torch.empty(1, dtype=tdt, device="cuda")
        k(a, out)
        print(f"{dtype} max clear={clear} nan_propagate=True -> {out.item()}"
              f"  {'OK' if torch.isnan(out).item() else 'BUG (expected nan)'}")
```

Output on L40S (sm_89), tilelang 0.1.11 (deterministic across two runs):

```
float16  max clear=True  nan_propagate=True -> nan   OK        <- positive control (MakeReduce path)
float16  max clear=False nan_propagate=True -> -1.0  BUG (expected nan)
bfloat16 max clear=True  nan_propagate=True -> nan   OK
bfloat16 max clear=False nan_propagate=True -> -1.0  BUG (expected nan)
```

`reduce_min` (NaN at index 13) behaves identically: `clear=True` → NaN, `clear=False` → `-1.0`. `nan_propagate=False` on both paths returns the plain non-NaN result (max=63.0, min=0.0/-1.0), as documented — so the defect is specifically `clear=False` **+** `nan_propagate=True`.

## Environment

- TileLang 0.1.11 (repro), source pinned at `cd37ed5f` (v0.1.11)
- L40S, sm_89, CUDA 12.x
- Compiled with `tilelang.compile`, cache disabled, private `TILELANG_CACHE_DIR`/`TMPDIR`

## Traceback

No traceback — compiles and runs, result silently wrong.

---

**Root cause.** `MakeReduce` (the per-thread/warp combine) honors `nan_propagate` and emits `__hmax_nan`/`__hmin_nan`, so the NaN survives the warp reduction. But `MakeUpdate` (the `clear=False` write-back merge into `dst_buffer`) emits a plain `Max`/`Min` (`__hmax`/`__hmin`) and never reads `op.nan_propagate`, so the final merge discards the NaN (`NaN < -1.0` is false → `-1.0` kept). The emitted CUDA source contains **both** `__hmax_nan` (correct MakeReduce) and a plain `__hmax(` (buggy MakeUpdate), which is the smoking gun.

<details>
<summary>Mechanism (pinned to cd37ed5f)</summary>

- [`MakeReduce` L163-177](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/backend/common/op/reduce.h#L163-L177): `use_nan_op = op.nan_propagate && (fp16 || bf16)`; for max/min/absmax emits `tl::max_nan()`/`tl::min_nan()` when set. Correct.
- [`MakeUpdate` L278-292](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/backend/common/op/reduce.h#L278-L292):
  ```cpp
  } else if (op.type->isMax() || op.type->isAbsMax()) {
    return Max(dst_val, src_val);   // L289 — plain Max, no nan branch
  } else if (op.type->isMin()) {
    return Min(dst_val, src_val);   // L291 — plain Min, no nan branch
  ```
  `MakeUpdate` never reads `op.nan_propagate`, so the propagated NaN is dropped on merge.
- `MakeUpdate` is invoked on the `clear=False` (`need_update`) path: [L744-745](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/backend/common/op/reduce.h#L744-L745) (batch>1) and [L817-818](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/backend/common/op/reduce.h#L817-L818) (scalar), gated by `need_update`, which is set for max/min/absmax when `!op.clear` ([L370-384](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/backend/common/op/reduce.h#L370-L384)).

**Fix:** in `MakeUpdate`, for the max/min/absmax branches emit the NaN variant (`tl::max_nan()`/`tl::min_nan()`) when `op.nan_propagate && (fp16 || bf16)`, mirroring `MakeReduce`'s `use_nan_op`.
</details>

**Provenance.** The runtime test [`test_reduce_max/min_runtime_nan_behavior`](https://github.com/tile-ai/tilelang/blob/cd37ed5f/testing/python/language/test_tilelang_language_reduce_maxmin_nan.py#L76-L103) asserts `nan_propagate=True` returns NaN, but its factory `_make_reduce_kernel` ([L19-30](https://github.com/tile-ai/tilelang/blob/cd37ed5f/testing/python/language/test_tilelang_language_reduce_maxmin_nan.py#L19-L30)) calls `reduce_fn(frag, out_frag, nan_propagate=nan_propagate)` with no `clear=` argument ([L27](https://github.com/tile-ai/tilelang/blob/cd37ed5f/testing/python/language/test_tilelang_language_reduce_maxmin_nan.py#L27)), so it always uses the default `clear=True` (`MakeReduce`) path. The `clear=False`/`MakeUpdate` merge is never exercised by the test suite.

**Dedup.** Distinct from all filed reduce issues — #2524 (batch>1 workspace aliasing), #2530 (keepdim ICHECK), #2542 (non-pow2 lane drop), #2568 (bool `MakeInitValue` INF overflow), #2570 (bitwise-float LHS ICHECK), #2583 (TMA int16 reduce), #2599 (warp_reduce register builtin) all have different roots. No filed issue or draft mentions `nan_propagate`/`MakeUpdate`/`hmax_nan`/nan-reduce.

**Reach.** Any `T.reduce_max`/`T.reduce_min`/absmax with `clear=False` (accumulate into a pre-existing output) plus `nan_propagate=True` on fp16/bf16 — a documented, non-error-path configuration. Silent wrong value, no diagnostic.
